backend/src/akvo/lumen/component/keycloak.clj

```diff
@@ -1,37 +1,75 @@
 (ns akvo.lumen.component.keycloak
   "We leverage Keycloak groups for tenant partition and admin roles.
    More info can be found in the Keycloak integration doc spec."
-  (:require [akvo.lumen.lib :as lib]
-            [akvo.lumen.protocols :as p]
-            [cheshire.core :as json]
-            [clj-http.client :as client]
-            [integrant.core :as ig]
-            [clojure.spec.alpha :as s]
-            [clojure.set :as set]
-            [clojure.tools.logging :as log]
-            [ring.util.response :refer [response]]))
+  (:require
+   [akvo.lumen.lib :as lib]
+   [akvo.lumen.protocols :as p]
+   [aleph.http :as http]
+   [byte-streams :as bs]
+   [cheshire.core :as json]
+   [clj-http.client :as client]
+   [clojure.set :as set]
+   [clojure.spec.alpha :as s]
+   [clojure.tools.logging :as log]
+   [integrant.core :as ig]
+   [manifold.deferred :as d]
+   [manifold.stream :as stream]
+   [ring.util.response :refer [response]])
+  (:import (io.aleph.dirigiste IPool)))
+
+
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Helper fns
 ;;;
 
 (defn fetch-openid-configuration
   "Get the openid configuration"
-  [issuer]
-  (-> (client/get (format "%s/.well-known/openid-configuration" issuer))
-      :body json/decode))
+  ([issuer]
+   (let [pool (http/connection-pool {:connection-options {:raw-stream? true}})]
+     (fetch-openid-configuration issuer pool)))
+  ([issuer pool]
+   (-> @(http/get (format "%s/.well-known/openid-configuration" issuer)
+                  {:pool pool})
+       :body
```

Comment:
Why not add a timeout here also?

Reply:
this is called only on system/start, 
currently for each entrypoint we defined in config.edn, so sometimes it takes a few time to make the same call several times, so i wrote it in a way it's called only once ... https://github.com/akvo/akvo-lumen/pull/2222/commits/638f0b678ebb31dd2b28239b0561a819f885e7c6

Reply:
Without the openid-configuration we can't really operate and should exit the process (log fatal?) and let K8s do a restart, or way better have retry logic and in worst case exit.

Reply:
Let the process die and k8s restart it until it works. We should already get an alert if this happens.

Reply:
tested that if we don't have a 20x status response then we get a proper ExceptionInfo :) 

```
ExceptionInfo clj-http: status 404 {:cached nil, :request-time 55, :repeatable? false, :protocol-version {:name "HTTP", :major 1, :minor 1}, :streaming? false, :http-client #object[org.apache.http.impl.client.InternalHttpClient 0x3ae6f731 "org.apache.http.impl.client.InternalHttpClient@3ae6f731"], :chunked? false, :type :clj-http.client/unexceptional-status, :reason-phrase "Not Found", :headers {"Connection" "keep-alive", "Content-Length" "0", "Date" "Wed, 17 Jul 2019 10:19:29 GMT"}, :orig-content-encoding nil, :status 404, :length 0, :body "", :trace-redirects []}

```